### PR TITLE
Adjust fetcher for new primary ip pricing format and ignore v6 primary addresses. (fixes #39)

### LINKED ISSUE #39: Error with the new IP pricing
Hey there, I just updated the container to the newest version and unfortunately it crashes with the following error:
```
goroutine 1 [running]:
github.com/jangraefen/hcloud-pricing-exporter/fetcher.Fetchers.MustRun(...)
	/app/fetcher/fetcher.go:97
main.main()
	/app/main.go:62 +0xbae
panic: 1 error(s) occurred:
* no primary IP pricing found for datacenter fsn1-dc14
```

Maybe this only affects instances in fsn1-dc14?

(btw thank you so much for this tool, it worked great for me in the last months)

### PR DESCRIPTION
**Pull request by byReqz**
I have adjusted the the fetcher for the new API format. Please consider this blocked until https://github.com/hetznercloud/hcloud-go/pull/222 is resolved.

### CODE REVIEW

File: fetcher/primaryip.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ func (primaryIP primaryIP) Run(client *hcloud.Client) error {
 	for _, p := range primaryIPs {
 		datacenter := p.Datacenter
 
-		hourlyPrice, monthlyPrice, err := primaryIP.pricing.PrimaryIP(p.Type, datacenter.Name)
+		hourlyPrice, monthlyPrice, err := primaryIP.pricing.PrimaryIP(p.Type, datacenter.Location.Name)
```
Comment (jangraefen) [inline fetcher/primaryip.go:27]: This is the correct change, I guess. I was wondering anyway, why Primary IP prices are based on datacenter and not on location, like everything else. Could you also adopt the `NewPrimaryIP` constructor to reflect that? Currently the Prometheus label is still called "datacenter", which should now be "location", like the others.

File: fetcher/prices.go [hunk 2]
```diff
@@ -40,6 +40,11 @@ func (provider *PriceProvider) PrimaryIP(ipType hcloud.PrimaryIPType, datacenter
...
 	defer provider.pricingLock.RUnlock()
 
+	// v6 pricing is not defined by the API
+	if string(ipType) == "ipv6" {
+		return 0, 0, nil
+	}
+
 	for _, byType := range provider.getPricing().PrimaryIPs {
 		if byType.Type == string(ipType) {
 			for _, pricing := range byType.Pricings {
```
Comment (jangraefen) [inline fetcher/prices.go:50]: After I looked at your PR on the hcloud-go repository, this makes sense, but I am actually quit suprised that the `Datacenter` attribute is now actually the location. To me, this is counter-inuitive. I would at least refactor this method to consistently call it location and add a comment why its then set to `p.Datacenter` 🙂.

**Review by jangraefen — CHANGES_REQUESTED** [review]
Very nice change, thank you so much! 😄
I just made some minor remarks, regarding naming.

**Review by jangraefen — APPROVED** [review]
Looks good to me! Thank you for you contributen ❤️.

I subscribed to the issue on the hcloud-go repository and will merge this PR as soon as the base issue is resolved.

**Comment by byReqz** [comment]
Would it make sense to also merge the individual DC labels at https://github.com/jangraefen/hcloud-pricing-exporter/blob/077338b90e641fb51070864bbba5a2d69d0f0156/fetcher/primaryip.go#L32-L33 
to just be the location? Not sure if thats stylistically in-line with the other stuff.

**Comment by jangraefen** [comment]
> Would it make sense to also merge the individual DC labels at
> 
> https://github.com/jangraefen/hcloud-pricing-exporter/blob/077338b90e641fb51070864bbba5a2d69d0f0156/fetcher/primaryip.go#L32-L33
> 
> to just be the location? Not sure if thats stylistically in-line with the other stuff.

While it would be possible to create the set of labels once and re-use it in the two calls, personally I don't think it is worth it. I am open for a refactoring though, but it should be the same for all fetchers then.

**Comment by jangraefen** [comment]
Also letting you know: I added a type label to both floating and primary IPs in a PR here: #41
I will merge it after your PR, to prevent a merge conflict for you 🙂.

**Comment by byReqz** [comment]
Updated the changes for the new proper `hcloud.PrimaryIPTypePricing.Location` field

**Comment by jangraefen** [comment]
Thanks for your contribution @byReqz. It is very much appreciated 🙂

**Pull request merged**